lib/taskana-core-test/src/test/java/acceptance/classification/query/QueryClassificationAccTest.java

```diff
@@ -0,0 +1,984 @@
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestFactory;
+import org.junit.jupiter.api.TestMethodOrder;
+import org.junit.jupiter.api.function.ThrowingConsumer;
+
+import pro.taskana.classification.api.ClassificationCustomField;
+import pro.taskana.classification.api.ClassificationService;
+import pro.taskana.classification.api.models.Classification;
+import pro.taskana.classification.api.models.ClassificationSummary;
+import pro.taskana.common.api.TimeInterval;
+import pro.taskana.common.internal.util.Pair;
+import pro.taskana.testapi.TaskanaInject;
+import pro.taskana.testapi.TaskanaIntegrationTest;
+import pro.taskana.testapi.security.WithAccessId;
+
+/**
+ * Acceptance test for all "get classification" scenarios.
+ *
+ * <p>Note: Instead of working with a clean database for each test, we create a scenario. In that
+ * scenario we use {@linkplain ClassificationCustomField#CUSTOM_1} to filter out all the other
+ * created {@linkplain Classification Classifications}. Therefore, all tests regarding {@linkplain
+ * ClassificationCustomField ClassificationCustomFields} are executed first.
+ *
+ * <p>The alternative is to create a "big" scenario as a test setup. But that would result to the
+ * same issues we have with the sql file.
+ */
+@TaskanaIntegrationTest
+@TestMethodOrder(value = MethodOrderer.OrderAnnotation.class)
+class QueryClassificationAccTest {
+  @TaskanaInject ClassificationService classificationService;
+
+  @WithAccessId(user = "businessadmin")
+  @TestFactory
+  @Order(1)
+  Stream<DynamicTest> should_FindClassifications_When_QueryingForCustomInFields() {
+    String testIdentifier = UUID.randomUUID().toString();
+
+    Iterator<ClassificationCustomField> customFieldIterator =
+        Arrays.stream(ClassificationCustomField.values()).iterator();
+    ThrowingConsumer<ClassificationCustomField> test =
+        customField -> {
+          ClassificationSummary classificationSummary =
+              defaultTestClassification()
+                  .customAttribute(customField, testIdentifier)
+                  .buildAndStoreAsSummary(classificationService);
+
+          List<ClassificationSummary> classificationSummaryList =
+              classificationService
+                  .createClassificationQuery()
+                  .customAttributeIn(customField, testIdentifier)
+                  .list();
+
+          ClassificationSummary masterClassificationSummary =
+              classificationService
+                  .getClassification(classificationSummary.getKey(), "")
+                  .asSummary();
+          assertThat(classificationSummaryList)
+              .containsExactlyInAnyOrder(classificationSummary, masterClassificationSummary);
+        };
+    return DynamicTest.stream(
+        customFieldIterator, c -> String.format("for %s", c.toString()), test);
+  }
+
+  @WithAccessId(user = "businessadmin")
+  @Test
+  void should_FindClassifications_When_QueryingForListValueField() throws Exception {
```

Comment:
What do you think about should_FindClassifications_When_QueryingUsingListValues() ?

Reply:
This test checks different test cases. Let's discuss creating a separate test for each test case.